Document/0x07a-Testing-Authentication-and-Session-Management.md

```diff
@@ -8,6 +8,65 @@ For all of the test cases below, it need to be investigated first what kind of a
 
 Cookie-Based Authentication is the traditional authentication mechanism used in web applications. In order to adopt to the different requirements of mobile apps Token-Based Authentication was specified and is more and more used nowadays in mobile apps. A prominent example for this is JSON Web Token or JWT<sup>[1]</sup> which can be part of an OAuth2 authentication and authorization framework.
 
+#### OAuth2
+
+OAuth2 is an authorization framework used to authorize an application to use a user account on an HTTP service for a limited time and, at the same time, preventing the client applications from having knowledge of any user credentials.
+
+OAuth2 defines 4 roles:
+
+* Resource Owner: the user owning the account.
+* Client: the application that wants to access the user's account using the access tokens.
+* Resource Server: hosts the user accounts.
+* Authorization Server: verifies the identity of the user and issues access tokens to the application.
```

Comment:
The Chicago Manual of Style (6.61) says: lowercase the first word unless it is a proper noun or the start of at least two complete sentences or a direct question.